py-polars/polars/series/series.py

```diff
@@ -958,11 +958,14 @@ def _arithmetic(self, other: Any, op_s: str, op_ffi: str) -> Self:
         if isinstance(other, pl.Expr):
             # expand pl.lit, pl.datetime, pl.duration Exprs to compatible Series
             other = self.to_frame().select_seq(other).to_series()
+        elif other is None:
+            other = pl.Series("", [None], dtype=self.dtype)
```

Comment:
I think it may be faster to create a Null series, as this has some fast paths for subsequent computation (Polars knows everything in the result must be null).

```suggestion
            other = pl.Series("", [None])
```

Reply:
Could have sworn I saw a case where it wanted to be typed appropriately, but after running through unit tests again I think I must have been hallucinating; committed the suggestion  :)

Reply:
If this doesn't work then there's an interesting bug we should fix with handling Null types, so let's see 😄